### PR TITLE
chore: Run mypy as CI job

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

File: auth/test/test_credentials.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def test_valid_oauth(app):
     assert result == ValidateResult(AuthKind.oauth, oauthtoken=oauth_token)
 
 
-def test_invalid_user(app):
+def test_invalid_password(app):
```
Comment (dmage) [inline auth/test/test_credentials.py:57]: the test `test_invalid_user` is defined at the line 123.

File: data/secscan_model/test/test_secscan_v4_model.py [hunk 2]
```diff
@@ -15,37 +15,33 @@
...
-from data.registry_model.datatypes import Manifest as ManifestDataType
 from data.registry_model import registry_model
 from util.secscan.v4.api import APIRequestFailure
-from util.canonicaljson import canonicalize
 from image.docker.schema2 import DOCKER_SCHEMA2_MANIFESTLIST_CONTENT_TYPE
 
 from test.fixtures import *
 
-from app import app, instance_keys, storage
+from app import app as application, instance_keys, storage
```
Comment (dmage) [inline data/secscan_model/test/test_secscan_v4_model.py:31]: `test.fixtures` provides a fixture named `app`, mypy gets confused when the `app` fixture is replaced with the `app` Flask object, it continues to think that `app` is a fixture.

File: mypy.ini [hunk 3]
```diff
@@ -1,8 +1,10 @@
 [mypy]
 python_version = 3.8
 mypy_path = mypy_stubs
-# local-dev is excluded until we decide what to do with __init__.py in those packages.
-exclude = local-dev/
+exclude = (?x)(
+    ^buildman/test/test_buildman\.py$ |
```
Comment (dmage) [inline mypy.ini:5]: The test seems to be broken:
```
buildman/test/test_buildman.py:11: error: Module "buildman.manager.ephemeral" has no attribute "REALM_PREFIX"
buildman/test/test_buildman.py:14: error: Module "buildman.server" has no attribute "BuildJobResult"
```

File: mypy.ini [hunk 4]
```diff
@@ -1,8 +1,10 @@
...
+    ^buildman/buildman_pb/buildman_pb2\.py$ |
```
Comment (dmage) [inline mypy.ini:6]: Autogenerated file, mypy doesn't fully understand it:
```
buildman/buildman_pb/buildman_pb2.py:1311: error: "GeneratedProtocolMessageType" has no attribute "BaseImage"
buildman/buildman_pb/buildman_pb2.py:1312: error: "GeneratedProtocolMessageType" has no attribute "GitPackage"
buildman/buildman_pb/buildman_pb2.py:1355: error: "GeneratedProtocolMessageType" has no attribute "PullMetadata"
```

File: mypy.ini [hunk 5]
```diff
@@ -1,8 +1,10 @@
...
+    ^endpoints/api/test/test_security\.py$ )
```
Comment (dmage) [inline mypy.ini:7]: Lots of wildcard imports confuse mypy:
https://github.com/quay/quay/blob/922a82a3d945002dd2d259d2d7950614249ef382/endpoints/api/test/test_security.py#L12-L38
```
endpoints/api/test/test_security.py:16: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:17: error: Name "datetime" already defined (by an import)
endpoints/api/test/test_security.py:20: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:22: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:23: error: Name "datetime" already defined (by an import)
endpoints/api/test/test_security.py:23: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:24: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:26: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:27: error: Incompatible import of "Repository" (imported name has type "Type[data.database.Repository]", local name has type "Type[endpoints.api.repository.Repository]")
endpoints/api/test/test_security.py:29: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:30: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:31: error: Name "model" already defined (by an import)
endpoints/api/test/test_security.py:32: error: Name "datetime" already defined (by an import)
endpoints/api/test/test_security.py:32: error: Incompatible import of "org_view" (imported name has type "Callable[[Any], Any]", local name has type "Callable[[Any, Any], Any]")
endpoints/api/test/test_security.py:33: error: Name "datetime" already defined (by an import)
endpoints/api/test/test_security.py:37: error: Name "datetime" already defined (by an import)
```

File: test/fixtures.py [hunk 6]
```diff
@@ -43,6 +41,9 @@
 INIT_DB_PATH = 0
 
 
+__all__ = ["init_db_path", "database_uri", "sqlitedb_file", "appconfig", "initialized_db", "app"]
```
Comment (dmage) [inline test/fixtures.py:44]: This package is typically used as `from test.fixtures import *`, without `__all__` it also provides `os`, `pytest`, `Flask`, `model`, etc.

File: test/test_api_usage.py [hunk 7]
```diff
@@ -3943,7 +3943,7 @@ def test_user_aggregate_logs(self):
         json = self.getJsonResponse(UserAggregateLogs)
         assert "aggregated" in json
 
-    def test_org_logs(self):
+    def test_org_aggregate_logs(self):
```
Comment (dmage) [inline test/test_api_usage.py:3946]: `test_org_logs` is defined at the line 3932.

File: _init.py [hunk 8]
```diff
@@ -7,7 +7,9 @@
 except ModuleNotFoundError:
     # Stub out this call so that we can run the external_libraries script
     # without needing the entire codebase.
-    def get_config_provider(*args, **kwargs):
+    def get_config_provider(
```
Comment (dmage) [inline _init.py:10]: We can remove `config_provider` from _init and move it to `app`, it'll eliminate this import.

File: workers/repomirrorworker/test/test_repomirrorworker.py [hunk 9]
```diff
@@ -2,6 +2,7 @@
 import mock
 import json
 from functools import wraps
+from mock import patch
```
Comment (flavianmissi) [inline workers/repomirrorworker/test/test_repomirrorworker.py:5]: can you import this `from unittest.mock import patch` instead?
`mock` is a dependency that we no longer need now that python's standard unittest package provides the same capabilities

File: requirements.txt [hunk 10]
```diff
@@ -40,7 +40,7 @@ futures==3.1.1
 geoip2==3.0.0
 gevent==21.8.0
 greenlet==1.1.2
-grpcio==1.30.0
+grpcio==1.46.3
```
[the remark this answers is not in the document]
Comment (flavianmissi) [reply requirements.txt:43]: what errors did you get? I'm able to install the requirements on macOS without problems 🧐

**Review by flavianmissi — COMMENTED** [review]
This is great, thank you for taking the time do it.
Left a couple questions, but otherwise LGTM.

File: requirements.txt (hunk 10 above)
[the remark this answers is not in the document]
Comment (dmage) [reply requirements.txt:43]: I had errors like `error: implicit declaration of function 'lseek' is invalid in C99` and it was fixed in later versions, see https://github.com/grpc/grpc/issues/25061

[the remark this answers is not in the document]
Comment (flavianmissi) [reply requirements.txt:43]: interesting! I'm fine with upgrading these packages - do you think we can trust that if the automated tests are passing the upgrade works? I'm unsure if the tests end up executing enough code.

File: workers/repomirrorworker/test/test_repomirrorworker.py (hunk 9 above)
Comment (dmage) [reply workers/repomirrorworker/test/test_repomirrorworker.py]: updated

File: requirements.txt (hunk 10 above)
[the remark this answers is not in the document]
Comment (dmage) [reply requirements.txt:43]: Automated tests for buildman are broken (see other comments in this PR). If it breaks buildman, we'll get motivation to fix them. But it should be backward compatible.

**Review by flavianmissi — APPROVED** [review]

**Comment by HammerMeetNail** [comment]
Gosh this is a large PR. 3.7.2 freeze is June 15th. When this gets merged, can we hold cherry picking to `redhat-3.7` until after release on June 20th?

**Comment by dmage** [comment]
I'd say it is 3.8 only.

**Pull request merged**